test/test_jsinterp.py

```diff
@@ -2,116 +2,68 @@
-        jsi = JSInterpreter('function f(){var x = [1,2,3]; x[0] = 4; x[0] = 5; x[2] = 7; return x;}')
-        self.assertEqual(jsi.call_function('f'), [5, 2, 7])
-
-    def test_parens(self):
-        jsi = JSInterpreter('function f(){return (1) + (2) * ((( (( (((((3)))))) )) ));}')
-        self.assertEqual(jsi.call_function('f'), 7)
-
-        jsi = JSInterpreter('function f(){return (1 + 2) * 3;}')
-        self.assertEqual(jsi.call_function('f'), 9)
-
-    def test_assignments(self):
-        jsi = JSInterpreter('function f(){var x = 20; x = 30 + 1; return x;}')
-        self.assertEqual(jsi.call_function('f'), 31)
-
-        jsi = JSInterpreter('function f(){var x = 20; x += 30 + 1; return x;}')
-        self.assertEqual(jsi.call_function('f'), 51)
-
-        jsi = JSInterpreter('function f(){var x = 20; x -= 30 + 1; return x;}')
-        self.assertEqual(jsi.call_function('f'), -11)
-
-    def test_comments(self):
-        'Skipping: Not yet fully implemented'
-        return
-        jsi = JSInterpreter('''
-        function x() {
-            var x = /* 1 + */ 2;
-            var y = /* 30
-            * 40 */ 50;
-            return x + y;
-        }
-        ''')
-        self.assertEqual(jsi.call_function('x'), 52)
-
-        jsi = JSInterpreter('''
-        function f() {
-            var x = "/*";
-            var y = 1 /* comment */ + 2;
-            return y;
-        }
-        ''')
-        self.assertEqual(jsi.call_function('f'), 3)
-
-    def test_precedence(self):
-        jsi = JSInterpreter('''
-        function x() {
-            var a = [10, 20, 30, 40, 50];
-            var b = 6;
-            a[0]=a[b%a.length];
-            return a;
-        }''')
-        self.assertEqual(jsi.call_function('x'), [20, 20, 30, 40, 50])
-
-    def test_call(self):
-        jsi = JSInterpreter('''
-        function x() { return 2; }
-        function y(a) { return x() + a; }
-        function z() { return y(3); }
-        ''')
-        self.assertEqual(jsi.call_function('z'), 5)
 
+class TestJSInterpreter(unittest.TestCase):
+    def setUp(self):
+        self.defs = defs
+
+
+def generator(test_case, name):
+    def test_template(self):
+        for test in test_case['subtests']:
+            jsi = JSInterpreter(test['code'], variables=None if 'globals' not in test else test['globals'])
+            if 'asserts' in test:
+                for a in test['asserts']:
+                    if 'call' in a:
+                        self.assertEqual(jsi.call_function(*a['call']), a['value'])
+                    else:
+                        self.assertEqual(jsi.run(), a['value'])
+            else:
+                log.debug('No asserts, skipping subtest')
+
+    log = logging.getLogger('TestJSInterpreter.%s' % name)
+
+    if 'i' not in test_case['skip']:
+        reason = False
+    else:
+        reason = test_case['skip']['i']
+
+    return test_template if not reason else unittest.skip(reason)(test_template)
```

Comment:
unittest.skip(reason)(test_template) looks counter-intuitive. Maybe raising SkipTest or calling skipTest() in the test function is better?